### ISSUE TITLE
Native CLI command-line tool using Graal native-image tool

### ISSUE THREAD

**Issue by msgilligan** [ConsensusJ/consensusj#34]
It would be very nice to be able to create a standalone, natively-compiled `cj-bitcoin-cli` command using [GraalVM](https://www.graalvm.org)

I started working on this and have a successful compile as of https://github.com/ConsensusJ/consensusj/commit/683a1f81bb6b4b9ab5b30c3575370f61eeb1675a

However, the executable fails as follows:

```
./bitcoinj-cli/build/cj-bitcoin-cli -rpcport 8080 getblockcount
Connecting to: http://127.0.0.1:8080/
[main] INFO org.bitcoinj.core.Context - Creating bitcoinj 0.14.7 context.
[main] INFO org.bitcoinj.core.Context - Creating bitcoinj 0.14.7 context.
Req json: {"method":"getblockcount","id":1,"jsonrpc":"1.0","params":[]}
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at java.lang.Throwable.<init>(Throwable.java:310)
	at java.lang.Exception.<init>(Exception.java:102)
	at java.lang.ReflectiveOperationException.<init>(ReflectiveOperationException.java:89)
	at java.lang.reflect.InvocationTargetException.<init>(InvocationTargetException.java:72)
	at com.oracle.svm.reflect.proxies.Proxy_133_BitcoinJCli_main.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:199)
	at Lcom/oracle/svm/core/code/CEntryPointCallStubs;.com_002eoracle_002esvm_002ecore_002eJavaMainWrapper_002erun_0028int_002corg_002egraalvm_002enativeimage_002ec_002etype_002eCCharPointerPointer_0029(generated:0)
Caused by: java.lang.AssertionError: Should not here
	at java.lang.Throwable.<init>(Throwable.java:265)
	at java.lang.Error.<init>(Error.java:70)
	at java.lang.AssertionError.<init>(AssertionError.java:58)
	at java.lang.AssertionError.<init>(AssertionError.java:74)
	at sun.util.logging.LoggingSupport.ensureAvailable(LoggingSupport.java:76)
	at sun.util.logging.LoggingSupport.isLoggable(LoggingSupport.java:114)
	at sun.util.logging.PlatformLogger$JavaLoggerProxy.isLoggable(PlatformLogger.java:656)
	at sun.util.logging.PlatformLogger.isLoggable(PlatformLogger.java:270)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1147)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1050)
	at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:984)
	at sun.net.www.protocol.http.HttpURLConnection.getOutputStream0(HttpURLConnection.java:1334)
	at sun.net.www.protocol.http.HttpURLConnection.getOutputStream(HttpURLConnection.java:1309)
	at org.consensusj.jsonrpc.RPCClient.send(RPCClient.java:100)
	at org.consensusj.jsonrpc.AbstractRPCClient.sendForResult(AbstractRPCClient.java:55)
	at org.consensusj.jsonrpc.AbstractRPCClient.send(AbstractRPCClient.java:114)
	at org.consensusj.jsonrpc.AbstractRPCClient.send(AbstractRPCClient.java:143)
	at com.msgilligan.bitcoinj.cli.BitcoinJCli.runImpl(BitcoinJCli.java:47)
	at com.msgilligan.bitcoinj.cli.CliCommand.run(CliCommand.java:164)
	at com.msgilligan.bitcoinj.cli.CliCommand.run(CliCommand.java:145)
	at com.msgilligan.bitcoinj.cli.BitcoinJCli.main(BitcoinJCli.java:31)
	... 4 more
```

It looks like the same issue was reported here: https://github.com/oracle/graal/issues/357 and fixed in: https://github.com/oracle/graal/commit/767d40bdbf8f1bd0b72022555878e7a8f3c88b2e but not yet released. Maybe the next release of GraalVM will have this fix.

**Comment by msgilligan** [comment]
To build the executable you must have GraalVM installed and the `$GRAAL_HOME` environment variable set. Then run:

    ./gradlew bitcoinj-cli:graalNativeImage

**Comment by cstancu** [comment]
@msgilligan yes, this fix will be included in the next release. In the meantime can you try and build from source (see [the quick start](https://github.com/oracle/graal/blob/master/substratevm/README.md#quick-start)) to check if there are any other issues.

**Comment by msgilligan** [comment]
Thanks @cstancu ! I'll probably wait until the next release to try it, but if I get anxious (and have the time) I'll try building from source.

**Comment by msgilligan** [comment]
@cstancu I tried `graalvm-ee-1.0.0-rc2` and the above problem seems to be resolved! Thanks again, for your help.

I am now able to successfully make a JSON-RPC call to a running `bitcoind` (well actually `omnicored` -- a modified version) and a get a result using `HttpURLConnection` and Jackson JSON serialization/deserialization.

**Comment by cstancu** [comment]
@msgilligan that's good to hear! Thanks for letting me know!

**Comment by msgilligan** [comment]
This is implemented in v0.4.0.
